### PR TITLE
DOC: add new page 'how to read these doc' in get started section

### LINKED ISSUE #39130: DOC: "getting started" under "coming from SQL" assumes you're using a notebook
#### Location of the documentation

https://pandas.pydata.org/docs/getting_started/comparison/comparison_with_sql.html

#### Documentation problem

I'm getting started with Pandas. I went to the docs site and found a section called "getting started" which sounded good to me. It then had a few boxes for where you're coming from and one of them was "coming from SQL" which also sounded good for me. I'm a back end dev. I was able to make a Python script (I'm using VS Code) that imported the library and ran the commands, but I saw no output. The docs showed output from `tips.head()`, but I saw no output from that line when I ran my script.

Googling my issue, I found this: https://stackoverflow.com/questions/51580863/output-of-pandas-head-function-is-not-displayed

The accepted answer suggested running `print(df.head())` instead of just `df.head()`. That worked for me. After reading that Stack Overflow page, I remembered that people also often use pandas with notebooks, so I went back to the docs page and looked at it closely. At that point, I noticed the `In [5]:` near the steps. I think people who've never used notebooks at all before might not even recognize this.

#### Suggested fix for documentation

- Remove the requirement to run the getting started code in a notebook, or
- Include alternative line for those running the script as is without viz tools like notebooks, or
- Add warning at top of getting started guide that readers are required to setup a notebook and run the code in it for their experience to match that of the guide

### PR DESCRIPTION
**Pull request by rhuille**
This page explains the code block are run in notebook

- [x] closes #39130
- [x] ~tests added / passed~
- [x] ~Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them~
- [x] ~whatsnew entry~

### CODE REVIEW

**Comment by rhuille** [comment]
Here is a screenshot of the new page: 

image

**Review by afeld — APPROVED** [review]
Looks good to me! I can imagine other things moving into this page (like the introductions to the sample data), but I think it's a good start. Approving, though I don't have merge rights.

File: doc/source/getting_started/how_to_read_these_docs.rst [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+or:
+
+.. ipython:: python
+
+    import pandas as pd
+    pd.DataFrame({'A': [1, 2, 3]})
+
+
+
+The first block is a standard python input, while in the second the ``In [1]:`` indicates the input is inside a `notebook <https://jupyter.org>`__. It does not make any difference **except the last line will be printed**.
```
Comment (MarcoGorelli) [inline doc/source/getting_started/how_to_read_these_docs.rst:26]: Hi @rhuille - I'm not sure about this "except the last line will be printed" part. Is that really a defining feature of Jupyter Notebooks? Even in a Python repl, you would see the last line printed. Perhaps mention that, in Jupyter Notebooks, plots will be shown inline (e.g. see #39527 where someone was confused by how to work with the visualisation docs)

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
instead of a separate page how about a note at the top of the getting started docs.

File: doc/source/getting_started/how_to_read_these_docs.rst (hunk 1 above)
Comment (rhuille) [reply doc/source/getting_started/how_to_read_these_docs.rst]: Hi @MarcoGorelli 
Thanks for your feedback, I agree it's not defining Jupyter Notebooks, so let's rewrite it like this: https://github.com/pandas-dev/pandas/pull/39207/commits/27c6bc6dc182f22c03f51b132b1259f3594ac8fa what do you think ?

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Hi @rhuille - could you address https://github.com/pandas-dev/pandas/pull/39207#issuecomment-779469318 please?

**Comment by mroeschke** [comment]
Thanks for the PR @rhuille but it appears this PR has gotten stale. Please let us know if you have time to merge master and to implement this as suggested in https://github.com/pandas-dev/pandas/pull/39207#issuecomment-779469318 and we'd be happy to reopen. Closing for now.

**Comment by ehallam** [comment]
Hi @mroeschke, I've picked up where @rhuille left off and I've added the changes to the top of the Getting Started page.
Does this look better? If so, I'll submit a new PR for my branch and reference it here.

PandasChanges

**Comment by mroeschke** [comment]
Thanks @ehallam this looks pretty good. One suggestion: Instead of putting this in `Getting Started`, I think it's would be more appropriate in https://pandas.pydata.org/docs/user_guide/index.html since that's where most of the code snippets are.

**Comment by ehallam** [comment]
Does this look good @mroeschke? I've also added a "Guides" heading to separate the "How to read these guides" section and the list of guides below.

PandasChanges4

**Comment by mroeschke** [comment]
Looks great! Looking forward to the PR 👍

**Pull request closed without merging**